api/v1/metric_jsonapi.go

```diff
@@ -0,0 +1,82 @@
+/*
+ *
+ * k6 - a next-generation load testing tool
+ * Copyright (C) 2021 Load Impact
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package v1
+
+import (
+	"time"
+
+	"go.k6.io/k6/stats"
+)
+
+type metricJSONAPI struct {
+	Data metricData `json:"data"`
+}
+
+// MetricsJSONAPI is JSON API envelop for metrics
+type MetricsJSONAPI struct {
+	Data []metricData `json:"data"`
+}
+
+type metricData struct {
+	Type       string `json:"type"`
+	ID         string `json:"id"`
+	Attributes Metric `json:"attributes"`
+}
+
+func newMetricEnvelope(m *stats.Metric, t time.Duration) metricJSONAPI {
+	return metricJSONAPI{
+		Data: newMetricData(m, t),
+	}
+}
+
+func newMetricsJSONAPI(list map[string]*stats.Metric, t time.Duration) MetricsJSONAPI {
+	metrics := make([]metricData, 0, len(list))
+
+	// TODO: lock ?
```

Comment:
I guess it could be a problem, is the pointer required?

Reply:
I believe here the pointers aren't required, but in general, it's `engine.Metrics`. And I see the lock there, but it used only for the unexported methods of the engine :thinking:

Reply:
Fixed with the latest commit